pandas/tests/io/test_pickle.py

```diff
@@ -563,11 +563,23 @@ def test_pickle_timeseries_periodindex():
     "name", [777, 777.0, "name", datetime.datetime(2001, 11, 11), (1, 2)]
 )
 def test_pickle_preserve_name(name):
-    def _pickle_roundtrip_name(obj):
-        with tm.ensure_clean() as path:
-            obj.to_pickle(path)
-            unpickled = pd.read_pickle(path)
-        return unpickled
 
-    unpickled = _pickle_roundtrip_name(tm.makeTimeSeries(name=name))
+    unpickled = _pickle_roundtrip(tm.makeTimeSeries(name=name))
     assert unpickled.name == name
+
+
+def test_pickle_datetimes(datetime_series):
+    unp_ts = _pickle_roundtrip(datetime_series)
+    tm.assert_series_equal(unp_ts, datetime_series)
+
+
+def test_pickle_strings(string_series):
+    unp_series = _pickle_roundtrip(string_series)
+    tm.assert_series_equal(unp_series, string_series)
+
+
+def _pickle_roundtrip(obj):
```

Comment:
can you remove in favor of tm.round_trip_pickle

Reply:
i thought about this but held off because they're not _quite_ the same.  tm.round_trip_pickle calls pd.to_pickle whereas this calls obj.to_pickle.  do we need to care about the difference?

Reply:
you are welcome to add direct pickle callas in round_trip_pickle

they are the same 

except that read_pickle can read from older versions

Reply:
will update

Reply:
Updated + greenish